assignments/arrays.js

```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
+        }
+    }
+};
+
+invSearch(inventory, 33);
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory.length -1;
+console.log(`The last car has id ${inventory[lastCar].id} and is a ${inventory[lastCar].car_year} ${inventory[lastCar].car_make} ${inventory[lastCar].car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+
+// const sortAlpha = function(arr){
+//     for(let i=0; i<arr.length; i++){
+//         carModels.push(arr[i].car_make);
+//     };
+//     retrun carModels; 
+// }
+
+// const modelSort = (arr) => {
+//         let modelList = [];
+//         for(let i = 0; i<arr; i++){
+//             modelList.push(arr[i].car_model);
+//         }
+//         return modelList;
+//     };
+//  
+// carModels = modelSort(inventory);
+
+// let carModels =[];
+// for( let i=0; i<arr.length; i++){
+//     for (prop in arr[i]){
+//         carModels.push(arr[i].car_model)
+//     };
+// }
+
+
+
+// let carModels = inventory.sort((a,b)=>{
+//     let modelA = a.car_model.toLowerCase();
+//     let modelB = b.car_model.toLowerCase();
+//     if(modelA < modelB){
+//         return -1; 
+//     }
+// });
 let carModels = [];
-console.log();
+
+const sortModels = (arr) => {
+    for(let i = arr.length -1; i >= 0; i--){
+        for(let j = 1; j<=i; j++){
+            if(arr[j-1].car_model>arr[j].car_model){
+                let temp = arr[j-1];
+                arr[j-1] = arr[j];
+                arr[j] = temp;
+            }
+        }
+    }
+    return arr;
+}
+
+// console.log(sortModels(inventory));
+
+
+
+
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+const findYears = (arr) => {
+    let yearArr = []
+    for(let i = 0; i < arr.length; i++){
+        yearArr.push(arr[i].car_year);
+    }
+    return yearArr;
+}
+
+carYears = findYears(inventory);
+// console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+
+const sortByAge = (arr, modelYear) => {
+    let arr2 = [];
+    for(let i = 0; i < arr.length; i++){
+        if(modelYear > arr[i].car_year){
+            arr2.push(arr[i]);
+        }
+    }
+    return arr2;
+}
+oldCars = sortByAge(inventory, 2000);
```

Comment:
this works but read the instruction again carefully, you already pushed all the car years into carYears array on line 154 and you were supposed to perform this challenge on that (would also save you a little time)